tools/topsites.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import base64
+import datetime
+import hashlib
+import hmac
+import os
+import requests
+import sys
+from urllib import urlencode
+from xml.dom.minidom import parseString
+
+from sqlalchemy import create_engine
+from sqlalchemy import Column
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from webcompat import app
+
+# Constant for alexa top site API
+ATS_ACTION_NAME = "TopSites"
+ATS_RESPONSE_GROUP_NAME = "Country"
+ATS_SERVICE_HOST = "ats.amazonaws.com"
+ATS_AWS_BASE_URL = "http://" + ATS_SERVICE_HOST + "/?"
+ATS_DATEFORMAT = "%Y-%m-%dT%H:%M:%S.%f"
+ATS_HASH_ALGORITHM = "HmacSHA256"
+ATS_COUNT = 100
+
+# Regions to dump to topsites.db
+REGIONS = ["GLOBAL", "US", "FR", "IN", "DE", "TW", "ID", "HK", "SG", "PL",
+           "GB", "RU"]
+
+# Alexa top site access and secret key
+ats_access_key = None
+ats_secret_key = None
+
+
+# Cache parsed sites, change priority if raised
+topsites = {}
+
+engine = create_engine("sqlite:///" + os.path.join(
+    app.config["BASE_DIR"], "topsites.db"))
+Base = declarative_base()
+Session = sessionmaker(bind=engine)
+session = Session()
+
+
+class Site(Base):
+    """SQLAchemy base object for an Alexa top site."""
+    __tablename__ = "topsites"
+
+    url = Column(String, primary_key=True)
+    priority = Column(Integer)
+    country_code = Column(String)
+    ranking = Column(Integer)
+
+    def __init__(self, url, priority, country_code, ranking):
+        self.url = url
+        self.priority = priority
+        self.country_code = country_code
+        self.ranking = ranking
+
+Base.metadata.create_all(engine)
+
+
+def query_topsites(country_code, count=1000):
+    """Query top sites with given country code and count."""
+    for index in range(1, count, 100):
+        uri = build_uri(country_code, index)
+        response = requests.get(uri)
+
+        if response.status_code == 200:
+            # See http://docs.aws.amazon.com/AlexaTopSites/latest/index.html?QUERY_QueryRequests.html  # nopep8
+            dom = parseString(response.content)
+            sites = dom.getElementsByTagName("aws:Site")
+            for site in sites:
+                parse_site(site)
+            session.commit()
+
+
+def parse_site(site):
+    """Parse given dom object."""
+    url = node_text(site, 'aws:DataUrl')
+    rank = int(node_text(site, 'aws:Rank'))
+
+    # Global website, initial priority is 1
+    # Tier 1 regions, initial priority is 2
+    priority = 1 if country_code == "" else 2
+
+    # Ranking between 100 and 1000, add priority 1
+    # Ranking between 1000 and 10000, add priority 2
+    if rank > 100 and rank <= 1000:
```

Comment:
I missed that in the first review. But python is so nice that it allows you to do this in one shot.

```python
>>> rank = 150
>>> if 100 < rank <= 1000:
...     print 'DOUBLE RAINBOW'
... 
DOUBLE RAINBOW
```

same for the next one.